R/utils-numeric.R

```diff
@@ -176,6 +176,104 @@ naToZero.default <- function(x) {
 }
 
 
+#' Drop columns containing all 0s
+#'
+#' This function drops all columns that have only 0 values. 
+#' @param x data.table, data.frame, list, or array
+#' @param ignoreNA boolean indicating if NA values should be ignored when determining all unique values in a column.
+#' Default is TRUE, which means a column that contains only 0 and NA values will be dropped. If FALSE, a column that contains 0s and
+#' NAs will not be dropped.
+#' @return cleanedData. A list that includes 1. "data" the result of removing some number of columns from the input x, and 2. "droppedColumns" names of the 
+#' columns that have been removed if the input x was a data.table, data.frame, or list. Otherwise "droppedColumns" contains
+#' the indicies of the input x that have been removed.
+#' @export
+dropZeroColumns <- function(x, ...) {
+  UseMethod("dropZeroColumns")
+}
+
+#' @export
+dropZeroColumns.data.table <- function(x, ignoreNA = TRUE) {
```

Comment:
there is an R standard for ignoring NA values. na.rm or rm.na or something.. maybe we should use that?